src/test/java/com/example/bank/userRegister/UserServiceTest.java

```diff
@@ -0,0 +1,157 @@
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            userService.registerUser(userRequest);
+        }, "User has to be adult.");
+    }
+
+    @DisplayName("A parameterized test of getUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldSource")
+    void shouldGetUserAndReturnUserFromId3(User user) throws UserNotFoundException {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.of(user));
+        User resultUser = userService.getUser(id);
+        Assertions.assertEquals(resultUser, user);
+    }
+
+    @DisplayName("A parameterized test of getUser() method")
+    @ParameterizedTest
+    @MethodSource("userSource")
+    void shouldReturnUserExceptionWhenThereIsNoUserWithId(User user) {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.empty());
+        Assertions.assertThrows(UserNotFoundException.class, () -> {
+            userService.getUser(id);
+        });
+    }
+
+    @DisplayName("A parameterized test of updateUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldUpdateSource")
+    void shouldReturnUpdatedUser(User user, UserRequest userRequest, User updatedUser) throws UserNotFoundException {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.of(user));
+        when(userRepository.save(user)).thenReturn(user);
+        User resultUser = userService.updateUser(userRequest, user.getId());
+        Assertions.assertEquals(updatedUser, resultUser);
+    }
+
+    @DisplayName("A parameterized test of updateUser() method")
+    @ParameterizedTest
+    @MethodSource("userOver18YearsOldSource")
+    void shouldThrowUserNotFoundExceptionDuringUserUpdateWhenThereIsNoUserWithId(User user, UserRequest userRequest) {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.empty());
+        Assertions.assertThrows(UserNotFoundException.class, () ->
+                userService.updateUser(userRequest, id));
+    }
+
+    @DisplayName("A parameterized test of updateUser() method")
+    @ParameterizedTest
+    @MethodSource("userBelow18YearsOldSource")
+    void shouldThrowIllegalArgumentExceptionDuringUserUpdateWhenTheUserIsNotAdult(User user, UserRequest userRequest) {
+        Long id = user.getId();
+        when(userRepository.findById(id)).thenReturn(Optional.of(user));
+        Assertions.assertThrows(IllegalArgumentException.class, () ->
+                userService.updateUser(userRequest, id));
+    }
+
+    @DisplayName("A parameterized test of deleteUser() method")
+    @ParameterizedTest
+    @MethodSource("userSource")
+    void shouldDeleteUserWhenGivenId(User user) throws UserNotFoundException {
+        /** Czy na tym etapie można jakoś łatwo zabezpieczyć metodę, żeby nie usuwała usera, jeżeli jego ID to null?? */
```

Comment:
czy ja dobrze rozumiem ze Ty chcesz w testach zabezpieczac metode? Czy ja to zle zrozumialem